File: src/main/java/org/funivan/lologa/algo/ga/genome/Genomes.java
package org.funivan.lologa.algo.ga.genome;

import org.funivan.lologa.algo.ga.player.PlayerInterface;

import java.util.ArrayList;
import java.util.Iterator;

public class Genomes implements Iterable<GenomeInterface> {
    private final Iterable<PlayerInterface> players;

    public Genomes(Iterable<PlayerInterface> players) {
        this.players = players;
    }

    @Override
    public Iterator<GenomeInterface> iterator() {
        ArrayList<GenomeInterface> result = new ArrayList<>();
        for (PlayerInterface player : this.players) {
            result.add(player.genome());
        }
        return result.iterator();
    }
}

File: src/main/java/org/funivan/lologa/algo/ga/genome/Genomes.kt
package org.funivan.lologa.algo.ga.genome

import org.funivan.lologa.algo.ga.player.PlayerInterface
import java.util.*

class Genomes(private val players: Iterable<PlayerInterface>) : Iterable<GenomeInterface> {

    override fun iterator(): Iterator<GenomeInterface> {
        val result = ArrayList<GenomeInterface>()
        for (player in this.players) {
            result.add(player.genome())
        }
        return result.iterator()
    }
}
